### ISSUE TITLE
Option to be able to supply a theme for Shiki

### ISSUE THREAD

**Issue by refineddigital** [TypeStrong/typedoc#1473]
## Search Terms

Theme, Shiki

## Problem

Our brand colours are basically a dark theme.  Typedoc is great at enabling you to create custom themes, but I don't see anywhere we can change the Shiki theme.  So when we render code it looks horrible.

![Screen Shot 2021-01-20 at 9 22 49 AM](https://user-images.githubusercontent.com/6423794/105088736-2d6ac900-5b01-11eb-9352-e77716b6abac.png)

## Suggested Solution

I see [here](https://github.com/TypeStrong/typedoc/blob/46371bfb1d932d2ed7f49aea855b4b9870c144d6/src/lib/utils/highlighter.ts#L28) on line 28 that it is using a fixed `light-plus`.  It would be good if it was possible to pass in the theme.
Maybe an option such as `highlight-theme` where you can pass in one of the themes.
Could be added to the typedoc.json file

**Comment by Gerrit0** [comment]
PR welcome! Should be fairly straightforward to implement. `highlight-theme` is good for now - but in 0.21 will need to be replaced with separate options for light and dark themes.

**Comment by refineddigital** [comment]
what is your thinking for `0.21`?
my thinking with `highlight-theme` option would be to pass one of the actual [Shiki themes](https://github.com/shikijs/shiki/blob/master/packages/themes/README.md#literal-values) through
```
export type Theme =
  | 'dark-plus'
  | 'github-dark'
  | 'github-light'
  | 'light-plus'
  | 'material-theme-darker'
  | 'material-theme-default'
  | 'material-theme-lighter'
  | 'material-theme-ocean'
  | 'material-theme-palenight'
  | 'min-dark'
  | 'min-light'
  | 'monokai'
  | 'nord'
  | 'slack-theme-dark-mode'
  | 'slack-theme-ochin'
  | 'solarized-dark'
  | 'solarized-light'
```

ie `--highlight-theme slack-theme-dark-mode` or via json `"highlight-theme": "slack-theme-dark-mode"`

**Comment by Gerrit0** [comment]
`--highlight-theme` seems good to me.

0.21 will require splitting this into `--highlight-theme-light` and `--highlight-theme-dark` since TypeDoc's default theme then will include both light and dark modes that users can choose between (much like Rust's docs https://doc.rust-lang.org/std/vec/struct.Vec.html)

**Comment by refineddigital** [comment]
@Gerrit0 I have just created a [pull request](https://github.com/TypeStrong/typedoc/pull/1486) for the initial `--highlightTheme` option.
This will clearly need to change in `0.21`. I'll be happy to port that over to `0.21` if you have a copy of that somewhere

**Comment by Gerrit0** [comment]
Awesome! Thanks :)

0.21 isn't in a state yet to need this, I'll merge it in once it is.

**Comment by Gerrit0** [comment]
Resolved by #1486, releasing in v0.20.20, in half an hour or so

**Comment by refineddigital** [comment]
Champion, thanks @Gerrit0
